nautobot_chatops/dispatchers/slack.py

```diff
@@ -208,16 +213,36 @@ def send_blocks(self, blocks, callback_id=None, modal=False, ephemeral=False, ti
             self.send_exception(slack_error)
 
     @BACKEND_ACTION_SNIPPET.time()
-    def send_snippet(self, text, title=None):
+    def send_snippet(self, text, title=None, ephemeral=None):
         """Send a longer chunk of text as a file snippet."""
+        if ephemeral is None:
+            ephemeral = settings.PLUGINS_CONFIG["nautobot_chatops"]["send_all_messages_private"]
+
         if self.context.get("channel_name") == "directmessage":
             channels = [self.context.get("user_id")]
         else:
             channels = [self.context.get("channel_id")]
         channels = ",".join(channels)
         logger.info("Sending snippet to %s: %s", channels, text)
         try:
-            self.slack_client.files_upload(channels=channels, content=text, title=title)
+            # Check for the length of the file if the setup is meant to be a private message
+            if ephemeral and len(text) < 40000:
+                self.slack_client.chat_postEphemeral(
+                    channel=self.context.get("channel_id"),
+                    user=self.context.get("user_id"),
+                    text="test",
+                    blocks=[
+                        {
+                            "type": "section",
+                            "text": {
+                                "type": "mrkdwn",
+                                "text": text
+                            }
+                        }
+                    ]
+                )
+            else:
+                self.slack_client.files_upload(channels=channels, content=text, title=title)
```

Comment:
The `len(text) < 40000` check seems like a potentially unwelcome surprise - if the chatbot is configured for `send_all_messages_private`, having messages over a certain length be unexpectedly sent as non-privately could be a security concern.

Maybe instead we could split the text into chunks and call `chat_postEphemeral` multiple times if needed?

Reply:
The toughest part here is that the likely place to have the split is on a large table. If that is the case we may need to look at handling headers appropriately. 

I'll think about this one some more and how we may be able to handle it.

Reply:
Any recommendations/thoughts on how to have the split occur? I'd like to some how do something with splitting on a `\n` character. Perhaps something with a look ahead in a function? I may have something on that later...